### DISCUSSION TITLE
How can you add a participant limit on the conference room?

### DISCUSSION THREAD

**Discussion by tremzo** [ant-media/Ant-Media-Server#5675] [FAQs]
If I create a conference room, I want to be able to set a limit on how many users can join the room. Where can I do this?

**Comment by tremzo** [comment]
What you can do is create a live stream and use that stream as the room when you join the conference. For example, if you create a new live stream either from the web panel or using the API and you name it "conference", then when you join a new conference room, make sure to join the room "conference". 

Then you can add a viewer limit on the newly created live stream that's called "conference" and this will set a participant limit.

**Reply by USAMAWIZARD** [reply]
Hi Tim I did not get how to do it can you please explain in a step by step manner Thank you

**Reply by tremzo** [reply]
Actually, we found that this method only limits the viewers at the conference, but those who get rejected, also start publishing. Here is the issue for that https://github.com/ant-media/Ant-Media-Server/issues/2898. 

The general process to restrict viewers is:

1. Create a new live stream using the API or web panel and edit the live stream to restrict the viewer count

2. When joining the conference, make sure to use the room name, which is the ID of the live stream you just created in step 1.
For example, in a multitrack conference, one streamId (mainTrack) is created with a room name that you joined or created. So you can create that mainTrack streamId in advance and put a WebRTC viewer limit on it.

3. Observe how a message is sent restricting access to view the conference when the participant limit is exceeded.
In a multitrack conference, when any user joins the room, the mainTrack plays on his or her end, which comprises other subTracks. So as the viewer limit is already set on the mainTrack streamId, a new participant cannot play other users streams, but his or her stream can still be published. For that, we have created the issue mentioned above.

While this works, as I mentioned, those who are rejected still end up publishing and so those currently in the conference end up viewing the participants who cannot join.

**Reply by USAMAWIZARD** [reply]
Thank  you @timantmedia
